### DISCUSSION TITLE
Integration with AWS Glue Schema Registry

### DISCUSSION THREAD

**Discussion by albanovito** [AbsaOSS/ABRiS#266] [General]
Hello guys, 

I read the documentation but I not found nothing. Is it possible to use this library with AWS Glue schema registry?

Thanks

**Comment by kevinwallimann** [comment]
Hi @albanovito No, ABRiS is only designed to work with Confluent Schema Registry. Currently, we have no plans of adding support for AWS Glue Schema Registry, but if you would like to fork or contribute, I suggest you look at [AbrisRegistryClient](https://github.com/AbsaOSS/ABRiS/blob/master/src/main/scala/za/co/absa/abris/avro/registry/AbrisRegistryClient.scala) and see how you can adapt it for AWS Glue Schema Registry.

**Reply by albanovito** [reply]
Thank you for your reply!!

**Reply by codeworm2** [reply]
Same request here for integration with AWS Glue Schema Registry.  The [AbrisRegistryClient](https://github.com/AbsaOSS/ABRiS/blob/master/src/main/scala/za/co/absa/abris/avro/registry/AbrisRegistryClient.scala) has import io.confluent.kafka.schemaregistry.client.SchemaMetadata in two methods.
@kevinwallimann  any idea to hide these confluent data structures here? 
The config object has many functions and classes return a Map[String, Any] https://github.com/AbsaOSS/ABRiS/blob/master/src/main/scala/za/co/absa/abris/config/Config.scala. 
Could you provide some constant string Map examples for different usages? I plan to use Java to call the from_avro function.

**Reply by kevinwallimann** [reply]
Hi @codeworm2. It is not possible to use AWS Glue Schema Registry with the current version of Abris, and we are currently not planning on implementing it. Abris is tightly integrated with the Confluent Schema Registry, because it uses its `SchemaMetadata` data structure internally (in SchemaManager, Config, possibly other classes as well)

If you would like to contribute to Abris, and implement the support for AWS Glue Schema Registry, you would have to create an abstraction around the `SchemaMetadata` data structure, and then provide implementations for both Confluent and AWS Glue Schema Registries. Additionally, you'd have to provide an implementation of the `AbrisRegistryClient` that supports AWS Glue Schema Registry.

**Reply by codeworm2** [reply]
Hi @kevinwallimann, Yes, my teammate and I want to contribute to ABRiS for the integration to AWS Glue Schema Registry. I branched the release 6.1.1 for compatibility with our production EMR environment.  

I agree that we need to abstract SchemaManager, and change some other code. Regarding the timeline, we can finish in Feb, hopefully. Would you mind if we have weekly zoom meeting to sync on the code chagnes? You can give me an email account, then I will invite you.

**Reply by kevinwallimann** [reply]
Hi @codeworm2 
Thank you for your invitation. I will happily review your pull request, but I think weekly zoom meetings are not necessary at this point.

**Reply by codeworm2** [reply]
Hi @kevinwallimann, our initial integration PoC works internally at AWS, but is not on prod yet. We will provide the diff in March. Sorry for the delay.

**Reply by roytjoo** [reply]
@codeworm2 Any update on your integration with AWS Glue Schema Registry?
